ssz_serialization/codec.nim

```diff
@@ -277,7 +279,89 @@ proc readSszValue*[T](input: openArray[byte],
     if resultBytesCount == maxExpectedSize:
       checkForForbiddenBits(T, input, val.maxLen + 1)
 
-  elif val is HashList | HashArray:
+  elif val is HashList:
+    type E = ElemType(type val)
+
+    when isFixedSize(E):
+      const elemSize = fixedPortionSize(E)
+      when elemSize > 1:
+        if input.len mod elemSize != 0:
+          var ex = new SszSizeMismatchError
+          ex.deserializedType = cstring typetraits.name(T)
+          ex.actualSszSize = input.len
+          ex.elementSize = elemSize
+          raise ex
+
+      val.data.setOutputSize input.len div elemSize
+      when supportsBulkCopy(type val.data[0]):
```

Comment:
```suggestion
      when supportsBulkCopy(ElementType(type val.data)):
```

afair we have a helper like that

Reply:
https://github.com/status-im/nim-ssz-serialization/pull/66/commits/79b3a47cd8adff928af5570306b7898b4b065aee

Reply:
https://github.com/status-im/nim-ssz-serialization/pull/66/commits/79b3a47cd8adff928af5570306b7898b4b065aee

These all seem to be equivalent:
```nim
      static:
        doAssert E is type E
        doAssert (type E) is E
        doAssert supportsBulkCopy(type val.data[0]) == supportsBulkCopy(type E)
        doAssert supportsBulkCopy(type val.data[0]) == supportsBulkCopy(E)
```

Which can be seen in a small example too:
```nim
var a: int
type E = typeof(int)
static: doAssert E is typeof(E)
static:
  echo E
  echo typeof(E)
```

I can't find if/where the Nim manual specifies this, but at least this change doesn't have any Nim semantic difference in practice.